motor_control/throttle/button_control.ino

```diff
@@ -0,0 +1,36 @@
+#include "button_control.h"
+
+ButtonThrottle::ButtonThrottle(const int &pin, int timeToFull): Throttle(pin, true){
+    this->m_timeToFull = timeToFull;
+    m_isPressed = false;
+}
+
+int ButtonThrottle::mappingFunction(int val, m_map_type mapType){
```

Comment:
I would choose a better variable name for `val`, maybe something like `time_pressed` would be good here too.